### PR TITLE
Change `NODE_ENV === production` to `module.hot`

### LINKED ISSUE #301: AppContainer complains because NODE_ENV === 'development'
I’ve just updated react-hot-loader to 3.0.0b2 in a project of mine (it’s gr8) and ran into something funky.

`AppContainer` was complaining about "react-hot-loader/patch" not running on a server rendered build of my project. Initially I thought I had to wrap `<App />` in `AppContainer` only if `module.hot` was set to true, but then I did some light digging and saw that `AppContainer` conditionally loads `AppContainer.dev.js` or `AppContainer.prod.js` based on the value of `NODE_ENV`.

I was doing some server rendering debugging and obviously didn’t need hot reloading, but I did need `NODE_ENV` to be set to development so I could see error messages.

All that to say, is there a better way to tell if hot reloading is desired that `NODE_ENV`? I’m a real dummy when it comes to the magical mystery of hot reloading, but can you check for `!module.hot` instead of `process.env.NODE_ENV === 'production'`?

If checking `NODE_ENV` is the best way to go, I think an addendum to the “react-hot-loader/patch didn’t run” message would be useful… something about `NODE_ENV` being set to development.

### PR DESCRIPTION
**Pull request by meyer**
Fixes #301, but I’m not sure if this is 👌 .

### CODE REVIEW

File: src/babel/index.js [hunk 1]
```diff
@@ -30,7 +30,7 @@ module.exports = function plugin(args) {
   const { types: t } = args;
 
   // No-op in production.
-  if (process.env.NODE_ENV === 'production') {
+  if (!module.hot) {
```
Comment (gaearon) [inline src/babel/index.js]: This will not be true for the Babel plugin itself.

File: src/webpack/tagCommonJSExports.js [hunk 2]
```diff
@@ -2,7 +2,7 @@
 
 ;(function register() { // eslint-disable-line no-extra-semi
   /* react-hot-loader/webpack */
-  if (process.env.NODE_ENV !== 'production') {
+  if (module.hot) {
```
Comment (gaearon) [inline src/webpack/tagCommonJSExports.js]: Similarly, `module.hot` will not be true inside Webpack loader itself, only in app's code.

File: src/AppContainer.js [hunk 3]
```diff
@@ -2,8 +2,8 @@
 
 'use strict';
 
-if (process.env.NODE_ENV === 'production') {
-  module.exports = require('./AppContainer.prod');
-} else {
+if (module.hot) {
```
Comment (gaearon) [inline src/AppContainer.js]: Let’s leave both checks here. `if (module.hot) && process.env.NODE_ENV === 'production'`

File: src/patch.js [hunk 4]
```diff
@@ -2,8 +2,8 @@
 
 'use strict';
 
-if (process.env.NODE_ENV === 'production') {
-  module.exports = require('./patch.prod');
-} else {
+if (module.hot) {
```
Comment (gaearon) [inline src/patch.js]: Same here: let’s check both.

**Comment by calesce** [comment]
👍  Looks good to me, thanks!

**Pull request merged**